src/qml/controls/TextButton.qml

```diff
@@ -78,4 +82,12 @@ Button {
             root.clicked()
         }
     }
+
+    onVisualFocusChanged: {
+        if (visualFocus && (focusReason == Qt.TabFocusReason || focusReason == Qt.BacktabFocusReason)) {
```

Comment:
visualFocus implies focusReason == Qt.TabFocus. Any reason for the redundant check?

Reply:
thanks, seems like it can actually be simplified into something like

``` diff
diff --git a/src/qml/controls/ContinueButton.qml b/src/qml/controls/ContinueButton.qml
index a49864075..6c51b40bf 100644
--- a/src/qml/controls/ContinueButton.qml
+++ b/src/qml/controls/ContinueButton.qml
@@ -27,15 +27,15 @@ Button {
 
         states: [
             State {
-                name: "DEFAULT"
+                name: "DEFAULT"; when: !visualFocus
                 PropertyChanges { target: bg; color: Theme.color.orange }
             },
             State {
-                name: "HOVER"
+                name: "HOVER"; when: visualFocus
                 PropertyChanges { target: bg; color: Theme.color.orangeLight1 }
             },
             State {
-                name: "PRESSED"
+                name: "PRESSED"; when: clicked
                 PropertyChanges { target: bg; color: Theme.color.orangeLight2 }
             }
         ]
@@ -65,11 +65,4 @@ Button {
             root.clicked()
         }
     }
-    onVisualFocusChanged: {
-        if (visualFocus && (focusReason == Qt.TabFocusReason || focusReason == Qt.BacktabFocusReason)) {
-            root.background.state = "HOVER"
-        } else {
-            root.background.state = "DEFAULT"
-        }
-    }
 }
```

will update